### DISCUSSION TITLE
Files ... View, Upload and Download

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9136] [Help]
Since I'm using an external (My-) SQL datasource, and there's no support for BLOB's (yet), I'm looking for another way of sharing files through BudiBase. I know this can be done via url links, but I do not want the files to be publicly accessible. I'm looking at S3 object storage via AWS or MinIO, but before I go down this route, I would like to know whether it's possible to view, upload and download files this way. And if so, how? Thanks in advance.

**Comment by melohagan** [comment]
Hey @ecxs 

I tried out this custom component recently which I thought was quite cool: https://github.com/chungchunwang/Budibase-File-Upload

Maybe it will be of some use to you!

**Reply by ecxs** [reply]
Thanks @melohagan I'm going to check out this custom component, but I really think it would be really nice (if not indispensable) that files/attachments support on external SQL datasource will be native BB functionality ...
